### ISSUE TITLE
dangerous for beta / rc versions

### ISSUE THREAD

**Issue by tomByrer** [vscode-contrib/vscode-versionlens#37]
Thanks for your plugin!  Usually great, but will trash beta versions, which may crash programs due to changed APIs:

package.json:
```json
| 1.0.1
"extract-text-webpack-plugin": "2.0.0-beta.4",
| 1.14.0
"webpack": "2.1.0-beta.27"...
```
https://github.com/webpack/extract-text-webpack-plugin
 "extract text plugin for webpack 2
The API has changed since version 1. For the webpack 1"

& latest in Webpack v2x is v2.2.0-rc.3 (they have a strange scheme)
Though [SemVer](http://semver.org/#spec-item-11) doesn't have an example of this scenario, this is how the WebPack (kinda popular lib right now ;) ) precedence should be:
 	
1.13.3 < 1.14.0 < 2.1.0-beta.27 < 2.1.0-beta.28 < 2.2.0-rc.0 < 2.2.0-rc.3

Maybe [SemVish](https://github.com/megawac/SemVish) can help here?

**Comment by pflannery** [comment]
Turns out this is happening because versionlens currently looks at the `latest` stream for each package using the `npm view` command. 
Example: 
`npm view webpack version` outputs `1.14.0`

For us to see webpack prereleases we would have to use it's `beta` stream
`npm view webpack@beta version` which outputs `2.2.0-rc.3`

Semver seems to handle this well for webpack.

I think the problem here is determining what stream to use for each package because every package can have a different stream for their pre-releases etc...

`dist-tags` shows us what streams are setup for a package.
`npm view webpack dist-tags` reveals `{ latest: '1.14.0', beta: '2.2.0-rc.3' }`
`npm view bootstrap dist-tags` reveals `{ latest: '3.3.7', next: '4.0.0-alpha.6' }`

Maybe we can have an ordered list of common stream names specified in the config for people who want to subscribe to different streams other than the `latest` stream?

**Comment by pflannery** [comment]
or maybe better to have a map in the config to let the user specifiy what stream to use per package?

**Comment by jens-duttke** [comment]
What about simply offering a link to every streams? So that the user is able to switch between the streams with just a click and without the need to know which streams are available.

Currently I'm using the {{next}} stream of TypeScript (version 2.3), but as soon as version 2.3 is available in the {{latest}} stream, I'll switch back to this stream, as long as TypeScript 2.4 will not offer any features which I need. That would be easily possible, if all available streams would be shown.

**Comment by tomByrer** [comment]
Or maybe slightly more automated?  :

```
If package.json > latest then
    if package.json > beta then
        error "your version is newer than npm's beta, use anyway?"
    elseif package.json = beta then
        return // no change
    elseif package.json < beta then
        offer update to beta
    endif
elseif package.json > latest then
    offer update to latest
endif
```
?

**Comment by pflannery** [comment]
This is implemented in 0.14.0

![image](https://cloud.githubusercontent.com/assets/1727302/25671395/c913e674-3027-11e7-910e-51a17905215c.png)

You can see the details on the [CHANGELOG](https://github.com/vscode-contrib/vscode-versionlens/blob/master/CHANGELOG.md)
